packages/arb-token-bridge-ui/src/components/TransferPanel/TokenButton.tsx

```diff
@@ -73,17 +77,32 @@ export function TokenButton(): JSX.Element {
     tokenImportDialogProps.onClose(false)
   }
 
-  function importToken(address: string) {
+  async function importToken(address: string) {
     setTokenToImport(address)
-    openTokenImportDialog()
+
+    const waitForInput = openTokenImportDialog()
+    const [confirmed] = await waitForInput()
+
+    if (confirmed) {
+      if (
+        isWithdrawOnlyToken(address, l2.network.id) ||
+        isTransferDisabledToken(address, l2.network.id)
+      ) {
+        openWithdrawOnlyDialog()
```

Comment:
Tested, added LINK token to queryparams to trigger Import Token dialog in deposit mode. Since it's deposit mode, and LINK is in withdraw only list, it should block me from importing, yet in the preview link, it lets me add the token to Transfer panel.

Reply:
<img width="905" alt="image" src="https://github.com/OffchainLabs/arbitrum-token-bridge/assets/7558499/cfa5b9b0-9e1b-42b3-9051-a4501e72fb5f">